src/app/loadout-builder/processWorker/process.ts

```diff
@@ -0,0 +1,494 @@
+
+              if (!groupedSetsInTotalTier) {
+                groupedSets[totalTier] = {};
+                groupedSetsInTotalTier = groupedSets[totalTier];
+              }
+
+              const existingSetAtTier = groupedSetsInTotalTier[tiers];
+              if (existingSetAtTier) {
+                existingSetAtTier.sets.push({
+                  armor: armor.map((items) => items.map((item) => item.id)),
+                  statChoices,
+                  maxPower,
+                });
+                if (maxPower > existingSetAtTier.maxPower) {
+                  existingSetAtTier.firstValidSet = firstValidSet.map((item) => item.id);
+                  existingSetAtTier.maxPower = maxPower;
+                  existingSetAtTier.firstValidSetStatChoices = statChoices;
+                }
+              } else {
+                // First of its kind
+                groupedSetsInTotalTier[tiers] = {
+                  sets: [
+                    {
+                      armor: armor.map((items) => items.map((item) => item.id)),
+                      statChoices,
+                      maxPower,
+                    },
+                  ],
+                  stats: stats as {
+                    [statType in StatTypes]: number;
+                  },
+                  // TODO: defer calculating first valid set / statchoices / maxpower?
+                  firstValidSet: firstValidSet.map((item) => item.id),
+                  firstValidSetStatChoices: statChoices,
+                  maxPower,
+                };
+              }
+
+              setCount++;
+
+              if (setCount > RETURNED_ARMOR_SETS && groupedSets[lowestTier]) {
+                const lowestTierGroup = groupedSets[lowestTier];
+                const lowestTierStatMixes = Object.keys(lowestTierGroup);
+
+                if (lowestTierStatMixes.length) {
+                  // Stats are sorted by order so this perfers sets by stat order preference
+                  const sortedStatMixes = lowestTierStatMixes.sort((a, b) => b.localeCompare(a));
+
+                  const leasePreferredStatMix = sortedStatMixes[sortedStatMixes.length - 1];
+
+                  setCount -= 1;
+                  lowestTierGroup[leasePreferredStatMix].sets
+                    .sort((a, b) => a.maxPower - b.maxPower)
+                    .pop();
+
+                  // If there are no sets left remove the group
+                  if (!lowestTierGroup[leasePreferredStatMix].sets.length) {
+                    delete lowestTierGroup[leasePreferredStatMix];
+                  }
+                }
+
+                // Remove the whole tier if there is nothing left in it and recalculate the lowest tier
+                if (!Object.keys(groupedSets[lowestTier]).length) {
+                  delete groupedSets[lowestTier];
+                  lowestTier = parseInt(
+                    Object.keys(groupedSets).sort((a, b) => a.localeCompare(b))[0],
```

Comment:
Worth testing, but `localeCompare` is probably slower than a simple compare.

Reply:
I will look into it, I just have a habit of always using localeCompare on strings.